plugins/scaffolder-backend/src/lib/repo/disk.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import glob from 'glob';
+import * as fs from 'fs-extra';
+import { logger } from 'lib/logger';
+import { Template, RepositoryBase as Base } from '.';
+
+interface DiskIndexEntry {
+  contents: Template;
+  location: string;
+}
+
+export class DiskRepository implements Base {
+  private repository: Template[] = [];
+  private localIndex: DiskIndexEntry[] = [];
+
+  constructor(private repoDir = `${__dirname}/templates`) {
+    this.reindex();
+  }
+
+  public async list(): Promise<Template[]> {
+    return this.repository;
+  }
+
+  public async reindex(): Promise<void> {
+    this.localIndex = await this.index();
+    this.repository = this.localIndex.map(({ contents }) => contents);
+  }
+
+  public async prepare(templateId: string): Promise<string> {
+    const template = this.localIndex.find(
+      ({ contents }) => contents.id === templateId,
+    );
+
+    if (!template) {
+      throw new Error('Template no found');
```

Comment:
Hmm - I mean it would be probably sensible to have our own version of something like the GRPC error codes.

And then we can map those to HTTP response codes, or if needed GRPC codes later on.

Basically the GRPC standard spec covers a lot more than just 404. So if we throw like `Backstage.NOT_FOUND` or something and then map that in Express to 404 or something would be nice.

Reply:
I wouldn't say this particular one should be translated to 404, maybe I haven't read carefully enough yet but it sounds like a 400

Reply:
(Agreed that we need to think about error preparation though!)